android/app/src/main/java/com/chirpreactnative/RCTChirpConnectModule.java

```
@@ -149,20 +148,12 @@ public void onError(ChirpError setConfigError) {
      * Configure the SDK with a config string.
      */
     @ReactMethod
-    public void setConfig(String config, final Promise promise) {
+    public void setConfig(String config) {
 
-        chirpConnect.setConfig(config, new ConnectSetConfigListener() {
-
-            @Override
-            public void onSuccess() {
-                promise.resolve("Initialisation Success");
-            }
-
-            @Override
-            public void onError(ChirpError setConfigError) {
-                promise.reject("SetConfig Error", setConfigError.getMessage());
-            }
-        });
+        ChirpError error = chirpConnect.setConfig(config);
```

Same here, we can return the error rather then use onError callback